sidecar/sidecar.py

```diff
@@ -91,8 +113,19 @@ def main():
 
     config.load_incluster_config()
     print("Config for cluster api loaded...")
-    namespace = open("/var/run/secrets/kubernetes.io/serviceaccount/namespace").read()
-    watchForChanges(label, targetFolder, url, method, payload, namespace)
+    with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace") as f:
+        namespace = f.read()
+
+    
+    while True:
+        try:
+            watchForChanges(label, targetFolder, url, method, payload, namespace, folderAnnotation)
+        except ApiException as e:
```

Comment:
I still don't think this is a good idea ( https://github.com/kiwigrid/k8s-sidecar/pull/6 ) but what would you say if we use the status inside the ApiException( https://github.com/kubernetes-client/python/blob/master/kubernetes/client/rest.py#L291) to only filter out specific status codes (eg only 500) and raise the exception for all other codes?

Reply:
My problem with saying only 50X errors are excepted is that in some circumstances, the connection to the API server is severed (no status code would be returned) and I *do not* want my pods restarting when this happens.

Reply:
sorry - I realize you were probably just referring to "ApiException" and not ProtocolError. I'm totally fine with only excepting 500s' here, I'll make this change.